conda/switch_cuda_version.sh

```diff
@@ -17,11 +17,18 @@ mkdir -p /usr/local
 rm -fr /usr/local/cuda
 ln -s "$CUDA_DIR" /usr/local/cuda
 
+CUDA_VERSION_DOT=$(nvcc --version | sed -n 4p|cut -f5 -d" " | cut -f1 -d",")
+export CUDA_VERSION=${CUDA_VERSION_DOT/./}
 if [[ "$OSTYPE" == "msys" ]]; then
-    export CUDA_VERSION=`ls /usr/local/cuda/bin/cudart64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2`
-    export CUDNN_VERSION=`ls /usr/local/cuda/bin/cudnn64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2`
+    export CUDNN_VERSION=$(ls /usr/local/cuda/bin/cudnn64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2)
+
+    # we want CUDNN_VERSION=8.1 for CUDA 11.2, not just 8
+    if [[ "$CUDA_VERSION" == '112' ]]; then
+        CUDNN_MAJOR=`grep 'define CUDNN_MAJOR' $(find /usr/local/cuda/ -name cudnn_version.h) | cut -d' ' -f3`
+        CUDNN_MINOR=`grep 'define CUDNN_MINOR' $(find /usr/local/cuda/ -name cudnn_version.h) | cut -d' ' -f3`
+        export CUDNN_VERSION=$CUDNN_MAJOR.$CUDNN_MINOR
+    fi
 else
-    export CUDA_VERSION=$(ls /usr/local/cuda/lib64/libcudart.so.*|sort|tac | head -1 | rev | cut -d"." -f -3 | rev)
     export CUDNN_VERSION=$(ls /usr/local/cuda/lib64/libcudnn.so.*|sort|tac | head -1 | rev | cut -d"." -f -3 | rev)
```

Comment:
Addressing some Shellcheck warnings:
- [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155) Declare and assign separately to avoid masking return values.
- [SC2012](https://github.com/koalaman/shellcheck/wiki/SC2012) Use `find` instead of `ls` to better handle non-alphanumeric filenames.
```suggestion
    CUDNN_VERSION=$(find /usr/local/cuda/lib64/libcudnn.so.* | sort | tac | head -1 | rev | cut -d"." -f -3 | rev)
    export CUDNN_VERSION
```